File: src/main/java/com/dashboard/exceptions/ExceptionFactory.java
package com.dashboard.exceptions;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;

public class ExceptionFactory {

    public static ResponseEntity<?> EMPLOYEE_NOT_FOUND(Integer employeeID){
        return new ResponseEntity<>(new ApiError("Employee with ID " + employeeID + " is not found", HttpStatus.NOT_FOUND), HttpStatus.NOT_FOUND);
    }

    public static ResponseEntity<?> TASK_NOT_FOUND(Integer taskID){
        return new ResponseEntity<>(new ApiError("Task with ID " + taskID + " is not found", HttpStatus.NOT_FOUND), HttpStatus.NOT_FOUND);
    }

    public static ResponseEntity<?> INTERNAL_EXCEPTION(){
        return new ResponseEntity<>(new ApiError("Internal error happened on backend", HttpStatus.INTERNAL_SERVER_ERROR), HttpStatus.INTERNAL_SERVER_ERROR);
    }

    public static ResponseEntity<?> ATTENDANCE_NOT_FOUND(Integer attendanceID){
        return new ResponseEntity<>(new ApiError("Attendance with ID " + attendanceID + " is not found", HttpStatus.NOT_FOUND), HttpStatus.NOT_FOUND);
    }

    public static ResponseEntity<?> ANNOUNCEMENT_NOT_FOUND(Integer announcementID){
        return new ResponseEntity<>(new ApiError("Announcement with ID " + announcementID + " is not found", HttpStatus.NOT_FOUND), HttpStatus.NOT_FOUND);
    }
}

File: src/main/kotlin/exceptions/ExceptionFactory.kt
package exceptions

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity

object ExceptionFactory {
    fun EMPLOYEE_NOT_FOUND(employeeID: Int): ResponseEntity<*> {
        return ResponseEntity(
            ApiError("Employee with ID $employeeID is not found", HttpStatus.NOT_FOUND),
            HttpStatus.NOT_FOUND
        )
    }

    fun TASK_NOT_FOUND(taskID: Int): ResponseEntity<*> {
        return ResponseEntity(ApiError("Task with ID $taskID is not found", HttpStatus.NOT_FOUND), HttpStatus.NOT_FOUND)
    }

    fun INTERNAL_EXCEPTION(): ResponseEntity<*> {
        return ResponseEntity(
            ApiError("Internal error happened on backend", HttpStatus.INTERNAL_SERVER_ERROR),
            HttpStatus.INTERNAL_SERVER_ERROR
        )
    }

    fun ATTENDANCE_NOT_FOUND(attendanceID: Int): ResponseEntity<*> {
        return ResponseEntity(
            ApiError("Attendance with ID $attendanceID is not found", HttpStatus.NOT_FOUND),
            HttpStatus.NOT_FOUND
        )
    }

    fun ANNOUNCEMENT_NOT_FOUND(announcementID: Int): ResponseEntity<*> {
        return ResponseEntity(
            ApiError("Announcement with ID $announcementID is not found", HttpStatus.NOT_FOUND),
            HttpStatus.NOT_FOUND
        )
    }
}